### PR TITLE
Issue#1835 remix plugin 0.0.2

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
**Ready to merge**
Implement remix-plugin 0.0.2.
See : https://github.com/ethereum/remix-ide/issues/1835

### CODE REVIEW

File: src/app/editor/SourceHighlighters.js [hunk 1]
```diff
@@ -1,24 +1,22 @@
 'use strict'
 const SourceHighlighter = require('./sourceHighlighter')
 
-import { ApiFactory } from 'remix-plugin'
+import { EditorApi } from 'remix-plugin'
 
-class SourceHighlighters extends ApiFactory {
+const profile = {
+  displayName: 'source highlighters',
```
Comment (yann300) [inline src/app/editor/SourceHighlighters.js]: `displayName` and `name` are not necessary right? they will be overridden by the `super` of `EditorApi`

Comment (GrandSchtroumpf) [reply src/app/editor/SourceHighlighters.js]: No, only the methods, events and kind are going to be overrided.

Comment (yann300) [reply src/app/editor/SourceHighlighters.js]: oh ok, then I think we should have a default naming for API shared among different platform. That does not need to be completely typescripted but at least well documented.
i.e if a file system is usually accessible under `call('fileSystem', '<function>', [])` , others platform have to provide it under the same name `fileSystem`.

in a further enhancement, we could add a function for retrieving modules of a specific `kind`.

Comment (GrandSchtroumpf) [reply src/app/editor/SourceHighlighters.js]: Sounds fair, but how you would deal with `compiler` ? You'll have the same `kind` with the same `methods` and `events`, but the names are different (`solidity`, `vyper`, `lll`, ...).

Comment (yann300) [reply src/app/editor/SourceHighlighters.js]: That would only apply with the core API like file manager, editor, context, udapp

Comment (yann300) [reply src/app/editor/SourceHighlighters.js]: That would only apply with the core API like file manager, editor, context, udapp

Comment (yann300) [reply src/app/editor/SourceHighlighters.js]: That would only apply with the core API like file manager, editor, context, udapp

Comment (GrandSchtroumpf) [reply src/app/editor/SourceHighlighters.js]: So the core API would be the one that are unique ?
Like : 
- core : `fileSystem`, `editor`, `network`, `udapp`.
- extension : `compiler`, `test`, `analysis`, `vm` (?)

Comment (yann300) [reply src/app/editor/SourceHighlighters.js]: Why 'vm' ? 
Yes, The core API provides infrastructure for module/plugin to interact with.

Comment (GrandSchtroumpf) [reply src/app/editor/SourceHighlighters.js]: Well, I guess at some point we would like to support eWASM Virtual Machine for example.

**Pull request merged**